HeterogeneousCore/AlpakaCore/README.md

```diff
@@ -0,0 +1,403 @@
+# Alpaka algorithms and modules in CMSSW
+
+## Introduction
+
+This page documents the Alpaka integration within CMSSW. For more information about Alpaka itself see the [Alpaka documentation](https://alpaka.readthedocs.io/en/latest/).
+
+### Compilation model
+
+The code in `Package/SubPackage/{interface,src,plugins,test}/alpaka` is compiled once for each enabled Alpaka backend. The `ALPAKA_ACCELERATOR_NAMESPACE` macro is substituted with a concrete, backend-specific namespace name in order to guarantee different symbol names for all backends, that allows for `cmsRun` to dynamically load any set of the backend libraries.
+
+The source files with `.dev.cc` suffix are compiled with the backend-specific device compiler. The other `.cc` source files are compiled with the host compiler.
+
+The `BuildFile.xml` must contain `<flags ALPAKA_BACKENDS="1"/>` to enable the behavior described above.
+
+## Overall guidelines
+
+* Minimize explicit blocking synchronization calls
+  * Avoid `alpaka::wait()`, non-cached memory buffer allocations
+* If you can, use `global::EDProducer` base class
+  * If you need per-stream storage
+    * For few objects consider using [`edm::StreamCache<T>`](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkGlobalModuleInterface#edm_StreamCacheT) with the global module, or
+    * Use `stream::EDProducer`
+  * If you need to transfer some data back to host, use `stream::SynchronizingEDroducer`
+* All code using `ALPAKA_ACCELERATOR_NAMESPACE` should be placed in `Package/SubPackage/{interface,src,plugins,test}/alpaka` directory
```

Comment:
I noticed `HeterogeneousCore/AlpakaCore/interface/MakerMacros.h` does not follow this convention. I think it would be more clear to move it to `HeterogeneousCore/AlpakaCore/interface/alpaka`. @fwyzard if you agree, I can take care of it.

Reply:
Sure, go ahead.